### PR TITLE
Add a `commitment feerate` parameter to multifundchannel

### PR DESCRIPTION
**Pull request by niftynei**
This lets a caller of `multifundchannel` specify a separate feerate for the channel's initial commitment transaction. This separates it from the feerate used for the funding transaction.

Writing the test for this uncovered an off by one in how we were calculating signature lengths.

Note that this conforms with the interface for `openchannel_init`, the new RPC that will initiate a v2 channel open (which contains two separate feerates, one for the funding tx and another for the commitment txs)

### CODE REVIEW

File: bitcoin/tx.c [hunk 1]
```diff
@@ -806,10 +806,10 @@ size_t bitcoin_tx_output_weight(size_t outscript_len)
 	return weight;
 }
 
-/* We grind signatures to get them down to 71 bytes (+1 for sighash flags) */
+/* We grind signatures to get them down to 71 bytes */
 size_t bitcoin_tx_input_sig_weight(void)
 {
-	return 1 + 71 + 1;
+	return 1 + 71;
```
Comment (darosior) [inline bitcoin/tx.c:812]: I think you overlooked the length prefix.

length prefix + sig + sighash_flag

**Review by darosior — CHANGES_REQUESTED** [review]

File: bitcoin/tx.c (hunk 1 above)
Comment (niftynei) [reply bitcoin/tx.c:812]: sighash_flag is included in the 71 count, see commit message

Comment (darosior) [reply bitcoin/tx.c:812]: Oh, right!

**Review by rustyrussell — APPROVED** [review]
Ack a45b7a2da85d62b7fba7b66ab30bb619dcb14855

**Pull request merged**